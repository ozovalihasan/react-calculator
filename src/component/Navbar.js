import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <div className="navbar main">
    <ul className="navbar container">
      <h2 className="navbar title">Math Magicians</h2>
      <li>
        <Link to="/">Home</Link>
      </li>
      |
      <li>
        <Link to="/calculator">Calculator</Link>
      </li>
      |
      <li>
        <Link to="/quote">Quote</Link>
      </li>
    </ul>
  </div>
);
export default Navbar;
