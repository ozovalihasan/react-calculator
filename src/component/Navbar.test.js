import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { screen } from '@testing-library/react';
import Navbar from './Navbar';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('contains 3 links with different urls', () => {
  act(() => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
      container
    );
  });
  // expect(getByText("Click Me").href).toBe("https://www.test.com/")
  expect(container.textContent).toBe('Math MagiciansHome|Calculator|Quote');
  expect(screen.getByText('Home').href).toBe('http://localhost/');
  expect(screen.getByText('Calculator').href).toBe(
    'http://localhost/calculator'
  );
  expect(screen.getByText('Quote').href).toBe('http://localhost/quote');
});
