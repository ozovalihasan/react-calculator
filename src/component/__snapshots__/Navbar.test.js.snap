// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navbar main"
>
  <ul
    className="navbar container"
  >
    <h2
      className="navbar title"
    >
      Math Magicians
    </h2>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    |
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    |
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</div>
`;
